charts/tekton-pipelines/templates/common/tasks/sentry-release.yaml

```diff
@@ -0,0 +1,82 @@
+{{ if .Values.sentry.enabled }}
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sentry-release
+  namespace: {{ .Release.Namespace }}
+
+spec:
+  description: >-
+    The task prepares and deploys sentry release for projects, which exist in Sentry.
+
+  workspaces:
+    - name: source
+
+  resources:
+    inputs:
+      - name: app
+        type: git
+
+  params:
+    - name: project
+      type: string
+      description: name of the project, which component is deployed
+
+    - name: component
+      type: string
+      description: name of the application component, which is deployed
+
+    - name: environment
+      type: string
+      description: environment name of the app being built, i.e. dev/staging/prod
+
+    - name: sourcemaps_dir
+      type: string
+      description: name of the dir where frontend sourcemaps would be stored in workspace
+      default: "sourcemaps"
+
+  stepTemplate:
+    env:
+      - name: SENTRY_AUTH_TOKEN
+        valueFrom:
+          secretKeyRef:
+            name: {{ .Values.sentry.authTokenSecret }}
+            key: {{ .Values.sentry.authTokenSecretKey }}
+      - name: SENTRY_ORG
+        value: {{ .Values.sentry.org }}
+      - name: SENTRY_URL
+        value: {{ .Values.sentry.url }}
+
+  steps:
+    - name: release
+      image: {{ .Values.images.sentry_cli }}
+      imagePullPolicy: {{ .Values.imagePullPolicy }}
+      workingDir: $(resources.inputs.app.path)
+      script: |
+        #!/usr/bin/env sh
+        set +x
+
+        PROJECT_NAME="$(params.project)-$(params.component)"
+        PROJECT=$(sentry-cli projects list | awk -v col=4 '{print $col}' | grep ^${PROJECT_NAME}$)
+        if [[ -z "${PROJECT}" ]]; then
+          echo "Didn't find '${PROJECT_NAME}' project in Sentry. Skip this step."
+          exit 0
+        else
+          echo "Found '${PROJECT_NAME}' project in Sentry, proceed with release."
+        fi
+
+        VERSION=$(sentry-cli releases propose-version)
+        sentry-cli releases new -p ${PROJECT_NAME} ${VERSION}
+        sentry-cli releases set-commits --auto --ignore-missing -p ${PROJECT_NAME} ${VERSION}
+
+        # needs to upload source maps for frontend projects
+        if [[ -d "$(workspaces.source.path)/$(params.sourcemaps_dir)" ]]; then
+          echo "Upload source maps for 'frontend' project"
+          sentry-cli releases files -p ${PROJECT_NAME} ${VERSION} upload-sourcemaps $(workspaces.source.path)/$(params.sourcemaps_dir)
+        else
+          echo "That is not 'frontend' project, no source maps should be uploaded"
+        fi
+
+        sentry-cli releases deploys -p ${PROJECT_NAME} ${VERSION} new -e $(params.environment)
+        sentry-cli releases finalize -p ${PROJECT_NAME} ${VERSION}
```

Comment:
what finalize does?

Reply:
https://docs.sentry.io/product/cli/releases/#finalizing-releases

Reply:
Do we need to finalize commits in the develop branches?
also per their doc:
https://docs.sentry.io/product/releases/
A release is a version of your code deployed to an environment

but in our implementation - the ENVIRONMENT is not mentioned/used anywehere.

they have [this](https://docs.sentry.io/product/releases/setup/):
```sh
sentry-cli releases --org ORGANIZATION_SLUG deploys VERSION new -e ENVIRONMENT
```

Reply:
I added environment param here `-e $(params.environment)`